src/Dependencies/Collections/SegmentedDictionary`2.cs

```diff
@@ -983,20 +989,16 @@ public void TrimExcess(int capacity)
             _version++;
             Initialize(newSize);
 
-            Debug.Assert(oldEntries is not null);
-
             CopyEntries(oldEntries, oldCount);
         }
 
-        private void CopyEntries(Entry[] entries, int count)
+        private void CopyEntries(SegmentedArray<Entry> entries, int count)
         {
-            Debug.Assert(_entries is not null);
```

Comment:
Similar comment - unclear why this is needed for Roslyn build.

Reply:
➡️ Same as https://github.com/dotnet/roslyn/pull/72687#discussion_r1559582639